api/src/test/java/uk/gov/dhsc/htbhf/card/service/CardServiceTest.java

```diff
@@ -76,8 +74,27 @@ void shouldSuccessfullyCreateCard() {
         verify(restTemplate).postForEntity(CARDS_URI, request, CardResponse.class);
     }
 
+    @Test
+    void shouldSuccessfullyGetBalance() {
+        //Given
+        String cardId = "123465";
+        String balanceUrl = buildGetBalanceUrl(cardId);
+        CardBalance cardBalance = aValidCardBalance();
+        given(restTemplate.getForEntity(anyString(), any())).willReturn(new ResponseEntity<>(cardBalance, HttpStatus.OK));
 
-    private String buildCardUrl(String cardId) {
+        //When
+        CardBalance result = cardService.getBalance(cardId);
+
+        //Then
+        assertThat(result).isEqualTo(cardBalance);
+        verify(restTemplate).getForEntity(balanceUrl, CardBalance.class);
+    }
+
+    private String buildDepositFundsUrl(String cardId) {
         return CARDS_URI + "/" + cardId + "/deposit";
     }
+
+    private String buildGetBalanceUrl(String cardId) {
+        return CARDS_URI + "/" + cardId + "/balance";
```

Comment:
Same here, can we remove the duplication with above?

Reply:
Done.